test/units/galaxy/test_collection.py

```diff
@@ -0,0 +1,618 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os
+import pytest
+import re
+import tarfile
+import tempfile
+import time
+import uuid
+
+from io import BytesIO, StringIO
+from units.compat.mock import MagicMock
+
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+
+from ansible.cli.galaxy import GalaxyCLI
+from ansible.errors import AnsibleError
+from ansible.galaxy import collection
+from ansible.module_utils._text import to_text
+from ansible.utils import context_objects as co
+from ansible.utils.display import Display
+from ansible.utils.hashing import secure_hash_s
+
+
+@pytest.fixture(autouse='function')
+def reset_cli_args():
+    co.GlobalCLIArgs._Singleton__instance = None
+    yield
+    co.GlobalCLIArgs._Singleton__instance = None
+
+
+@pytest.fixture()
+def collection_input(tmp_path_factory):
+    test_dir = str(tmp_path_factory.mktemp('test-ÅÑŚÌβŁÈ Collections Input'))
+    namespace = 'ansible_namespace'
+    collection = 'collection'
+    skeleton = os.path.join(os.path.dirname(os.path.split(__file__)[0]), 'cli', 'test_data', 'collection_skeleton')
```

Comment:
Isn't it better readable with pathlib?

```suggestion
    skeleton = Path(__file__).parent / 'cli' / 'test_data' / 'collection_skeleton'
```

Reply:
Or
```suggestion
    skeleton = (Path(__file__) / '..' / 'cli' / 'test_data' / 'collection_skeleton').resolve()
```

Reply:
Since `pytest` already pulls in a backport for it, you could benefit from that:
```python
try:
    from pathlib import Path
except ImportError:
    from pathlib2 import Path
```